modules/util/config/CloudConfig.py

```diff
@@ -0,0 +1,81 @@
+from typing import Any
+
+from modules.util.config.BaseConfig import BaseConfig
+from modules.util.enum.CloudAction import CloudAction
+from modules.util.enum.CloudType import CloudType
+
+
+class CloudConfig(BaseConfig):
+    enabled: bool
+    type: CloudType
+    api_key: str
+    create : bool
+    name: str
+    id: str
+    tensorboard_tunnel: bool
+    sub_type: str
+    gpu_type: str
+    volume_size: int
+    min_download: int
+    host: str
+    port: int
+    user: str
+    huggingface_token: str
```

Comment:
Adding the huggingface_token to the config is a problem. Config files are regularly shared. If the token is included in that file, users can unknowingly expose it. That's also the reason I haven't added a setting for this yet. It needs to be saved in some different place. It would also be pretty useful to have this setting outside the cloud training environment.

I could imagine having a different user specific config file that's never shared or exported. But I haven't really put a lot of thought into it yet.

Reply:
same for runpod API key then

Reply:
asking AI for best practices for storing secrets in python applications, it lists environment variables, configuration files, and this:
![image](https://github.com/user-attachments/assets/0a976d33-ea03-421a-8dc9-ad2363d4d3d1)

what do you think?
would therefore be shared among all configs, not linked to a specific config

Reply:
I think it makes sense to use the same structure in the python code (derive a new class from BaseConfig, save that as a json file, use a UIState to display the values in the ui). The json file could be saved either in a new subdirectory or directly in the root directory. I don't really have a preference there.

Reply:
slightly different proposal here:
https://github.com/dxqbYD/OneTrainer/commit/08b5f646715cb03ffa6c37903accdd5470d92a0b

By keeping the secrets within the config hierarchy, you avoid having to pass around two different config objects wherever you might need a secret

secrets.json:
```{
    "__version": 6,
    "cloud": {
        "__version": 0,
        "api_key": "",
        "id": "",
        "huggingface_token": "",
        "host": "",
        "port": "0",
        "user": "root"
    }
}```